### DISCUSSION TITLE
Roadmap

### DISCUSSION THREAD

**Discussion by slyh** [slyh/Cities2-TrafficLightsEnhancement#6] [General]
**Short Term**

- [x] Option for always-green left turn (in LHT) / right turn (in RHT)
- [x] Rewrite the web UI
- [x] Lane management (retired)

**Long Term**

- [x] Implement a raycaster for better UX
- [x] Custom traffic light phases
- [ ] Leading / Trailing signals
- [ ] Timing management (flow-based)
- [ ] Traffic lights synchronisation
- [ ] Refactor code so that this project can be released under a licence

**Features Under Consideration**

- Copy and paste custom phase setups

**Comment by timfrombriz** [comment]
Hey Slyh, trying to reach out to you, in case my comment isnt seen here...
https://github.com/slyh/Cities2-Various-Mods/issues/11

Cheers.

**Comment by michaelberthold** [comment]
Hey @slyh I would be glad if timing management would be added. But I fear that it's not an easy task. Because of the rushhour simulation you either need different phases for certain daytimes or an intelligent phasing dependend on the number of cars wainting on each side of the intersection.

**Reply by slyh** [reply]
My current plan is to allow users to set a minimum phase length, with the option for more extensions if necessary. The built-in traffic lights are already smart enough to extend the signal as long as cars continue to flow.

**Comment by brewcrew87** [comment]
Hey @slyh - love this mod.  Thanks for putting in the work!

Quick question - do you think from what you have seen that it would be possible to have pedestrians wait to walk until the "straight ahead" phase starts instead of starting it during the protected left when that is on?  I love the protected left, but traffic is getting hamstrung all because pedestrians start to walk upon the left turn phase instead of waiting for the straight/right turn phase.

I understand it is probably a lot more difficult than it sounds, but figured I'd ask since I don't think I saw anyone else bring it up!

Thanks again for all of your work.

**Reply by slyh** [reply]
I think that is feasible.

**Comment by OutsideOctaves** [comment]
Quick question:  Do you plan on an official release on PDX Mods or just for Github and thunderstore?

**Reply by tqb333** [reply]
The only mod i wait.... and still no answer :(

**Reply by slyh** [reply]
For this question, please refer to issue #87.
